src/gui/dialogs/addon/addon_auth.hpp

```diff
@@ -0,0 +1,57 @@
+/*
+   Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include "gui/dialogs/modal_dialog.hpp"
+
+namespace gui2
+{
+class field_text;
+
+namespace dialogs
+{
+
+/**
+ * @ingroup GUIWindowDefinitionWML
+ *
+ * This shows the dialog to provide a password when uploading an add-on.
+ * 
+ * Key               |Type           |Mandatory|Description
+ * ------------------|---------------|---------|-----------
+ * password          | text_box      |yes      |The password used to verify the uploader.
+ */
+class addon_auth : public modal_dialog
+{
+public:
+	addon_auth();
+
+   std::string get_password();
+
+private:
+	/** Inherited from modal_dialog, implemented by REGISTER_DIALOG. */
+	virtual const std::string& window_id() const override;
+
+	/** Inherited from modal_dialog. */
+	virtual void pre_show(window& window) override;
+
+	/** Inherited from modal_dialog. */
```

Comment:
Doxygen generates better docs without this comment - without it, Doxygen will copy the docs from modal_dialog.

Reply:
This is a common problem with a lot of different dialogs then, and all should be fixed rather than just this one, which is also outside the scope of this PR.  The boost tests in your previous comment I don't really feel the need to address to be honest, but I'll push a separate commit to mass remove these comments after this PR is merged.